### PR TITLE
LWDEV-9284-rel Add swagger method to turn on/off maintenance mode

### PR DESCRIPTION
**Pull request by rocketmannot**

### CODE REVIEW

File: src/main/kotlin/com/lykke/matching/engine/config/spring/WebConfig.kt [hunk 1]
```diff
@@ -56,6 +60,14 @@ open class WebConfig  {
                 .validatorUrl(null)
                 .build()
     }
+    @Bean
+    open fun settingValidators(settingValidators: List<SettingValidator>): Map<AvailableSettingGroup, List<SettingValidator>> {
+        return settingValidators.groupBy { it.getSettingGroup() }
+    }
+
+    private fun gson(): Gson = GsonBuilder()
```
Comment (barsukovskii) [inline src/main/kotlin/com/lykke/matching/engine/config/spring/WebConfig.kt]: not used private method

File: src/main/kotlin/com/lykke/matching/engine/utils/order/MessageStatusUtils.kt [hunk 2]
```diff
@@ -52,6 +52,9 @@ class MessageStatusUtils {
                 ValidationException.Validation.BALANCE_LOWER_THAN_RESERVED -> MessageStatus.BALANCE_LOWER_THAN_RESERVED
                 ValidationException.Validation.NEGATIVE_OVERDRAFT_LIMIT -> MessageStatus.NEGATIVE_OVERDRAFT_LIMIT
                 ValidationException.Validation.LIMIT_ORDER_NOT_FOUND -> MessageStatus.LIMIT_ORDER_NOT_FOUND
+                ValidationException.Validation.NOT_ACCEPTABLE_MESSAGE_SWITCH_SETTING_VALUE -> MessageStatus.BAD_REQUEST
```
Comment (barsukovskii) [inline src/main/kotlin/com/lykke/matching/engine/utils/order/MessageStatusUtils.kt]: Remove this Validation value from here and from enum

File: src/main/kotlin/com/lykke/matching/engine/utils/order/MessageStatusUtils.kt [hunk 3]
```diff
@@ -52,6 +52,9 @@ class MessageStatusUtils {
...
+                ValidationException.Validation.GENERIC_VALIDATION_FAILURE -> MessageStatus.BAD_REQUEST
```
Comment (barsukovskii) [inline src/main/kotlin/com/lykke/matching/engine/utils/order/MessageStatusUtils.kt]: Duplicated value in the block

File: src/main/kotlin/com/lykke/matching/engine/services/validators/impl/ValidationException.kt [hunk 4]
```diff
@@ -16,6 +16,7 @@ class ValidationException(val validationType: Validation, validationMessage: Str
         UNKNOWN_ASSET("Unknown asset"),
         BALANCE_LOWER_THAN_RESERVED("Balance lower than reserved"),
         LIMIT_ORDER_NOT_FOUND("Limit order not found"),
+        NOT_ACCEPTABLE_MESSAGE_SWITCH_SETTING_VALUE("Supplied value is not supported"),
```
Comment (barsukovskii) [inline src/main/kotlin/com/lykke/matching/engine/services/validators/impl/ValidationException.kt]: https://github.com/LykkeCity/MatchingEngine/pull/179#discussion_r217014356

File: src/main/kotlin/com/lykke/matching/engine/config/spring/WebConfig.kt (hunk 1 above)
Comment (rocketmannot) [reply src/main/kotlin/com/lykke/matching/engine/config/spring/WebConfig.kt]: fixed

File: src/main/kotlin/com/lykke/matching/engine/utils/order/MessageStatusUtils.kt (hunk 3 above)
Comment (rocketmannot) [reply src/main/kotlin/com/lykke/matching/engine/utils/order/MessageStatusUtils.kt]: fixed

File: src/main/kotlin/com/lykke/matching/engine/utils/order/MessageStatusUtils.kt (hunk 2 above)
Comment (rocketmannot) [reply src/main/kotlin/com/lykke/matching/engine/utils/order/MessageStatusUtils.kt]: fixed

File: src/main/kotlin/com/lykke/matching/engine/services/validators/impl/ValidationException.kt (hunk 4 above)
Comment (rocketmannot) [reply src/main/kotlin/com/lykke/matching/engine/services/validators/impl/ValidationException.kt]: fixed

**Pull request merged**